File: chunker/formatting/removeHorizontalRules.ts
/**
 * Удаляет горизонтальные линии Markdown: строки, состоящие только из дефисов, звёздочек или подчёркиваний (с пробелами).
 * Например: ---, ***, ___
 */
export function removeHorizontalRules(text: string): string {
  return text.replace(/^\s*[-*_]{3,}\s*$/gm, "").trim();
}

File: src-tauri/src/chunker/formatting/remove_horizontal_rules.rs
use std::sync::LazyLock;
use regex::Regex;
use std::borrow::Cow;

static RE: LazyLock<Regex> = LazyLock::new(|| Regex::new(r"(?m)^(?:\*\*\*+|---+|___+)\s*$").unwrap());

pub fn remove_horizontal_rules(text: &str) -> Cow<'_, str> {
    RE.replace_all(text, "")
}
